node/src/config_files/chainstate/mod.rs

```diff
@@ -35,7 +34,7 @@ pub struct ChainstateConfigFile {
     ///
     /// The initial block download is finished if the difference between the current time and the
     /// tip time is less than this value.
-    pub max_tip_age: Option<u64>,
+    pub max_tip_age: Option<RpcDuration>,
```

Comment:
My problem with this is that ChainstateConfig has nothing to do with Rpc. This doesn't make sense except for the coincidental relationship that both use an integer (and notice the disaster if we're using f64 here, which is another story). The time in a configuration file or configuration input shouldn't be quantified by some arbitrary units unclear. Given that this is crossing the boundaies of the software and type-safety is not possible, the correct way to do this is to use an integer (like u64, like it was), and clarify what units it uses. Then, do the conversion to something that's type-safe that'll deal with this correctly.

Reply:
Sure, reverted.